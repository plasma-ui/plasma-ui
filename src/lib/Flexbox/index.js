import styled from "@emotion/styled";
import propTypes from "@styled-system/prop-types";
import {
  createShouldForwardProp,
  props
} from "@styled-system/should-forward-prop";
import {
  flexbox,
  space,
  layout,
  color,
  typography,
  grid,
  background,
  border,
  position,
  shadow,
  compose
} from "styled-system";

const shouldForwardProp = createShouldForwardProp([
  ...props,
  "dir",
  "justify",
  "align",
  "transform"
]);

const Flexbox = styled("div", {
  shouldForwardProp
})(
  props => ({
    flexDirection: props.dir,
    justifyContent: props.justify,
    alignItems: props.align,
    transform: props.transform
  }),
  {
    boxSizing: "border-box"
  },
  compose(
    flexbox,
    space,
    layout,
    color,
    typography,
    grid,
    background,
    border,
    position,
    shadow
  )
);

Flexbox.defaultProps = {
  display: "flex",
  dir: "row",
  align: "center"
};

Flexbox.propTypes = {
  ...propTypes.flexbox,
  ...propTypes.space,
  ...propTypes.layout,
  ...propTypes.color,
  ...propTypes.typography,
  ...propTypes.grid,
  ...propTypes.background,
  ...propTypes.border,
  ...propTypes.position,
  ...propTypes.shadow
};

export default Flexbox;
