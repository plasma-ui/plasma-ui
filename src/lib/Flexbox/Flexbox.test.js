import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

import Flexbox from "./";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Flexbox />, div);
});

it("renders correctly with matching snapshot", () => {
  const tree = renderer.create(<Flexbox />).toJSON();
  expect(tree).toMatchSnapshot();
});
