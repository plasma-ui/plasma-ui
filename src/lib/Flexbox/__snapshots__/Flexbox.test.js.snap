// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with matching snapshot 1`] = `
<div
  className="css-ju9zgm"
/>
`;
